src/lamp_py/ingestion/glides.py

```diff
@@ -107,31 +107,28 @@ def append_records(self) -> None:
                 while start < now:
                     end = start + relativedelta(months=1)
                     if end < now:
-                        unique_table = (
-                            pl.DataFrame(
-                                joined_ds.filter(
-                                    (pc.field("time") >= start)
-                                    & (pc.field("time") < end)
-                                ).to_table()
-                            )
-                            .unique(keep="first")
-                            .to_arrow()
-                            .sort_by("time")
+                        row_group = pl.DataFrame(
+                            joined_ds.filter(
+                                (pc.field("time") >= start)
+                                & (pc.field("time") < end)
+                            ).to_table()
                         )
+
                     else:
+                        row_group = pl.DataFrame(
+                            joined_ds.filter(
+                                (pc.field("time") >= start)
+                            ).to_table()
+                        )
+
+                    if row_group.is_empty() > 0:
                         unique_table = (
-                            pl.DataFrame(
-                                joined_ds.filter(
-                                    (pc.field("time") >= start)
-                                ).to_table()
-                            )
-                            .unique(keep="first")
+                            row_group.unique(keep="first")
+                            .sort(by=["time"])
                             .to_arrow()
-                            .sort_by("time")
+                            .cast(new_dataset.schema)
```

Comment:
i was missing this cast before, which caused incompatible schema errors. (it seems `to_arrow()` makes all strings large strings.